src/logic/wallets/getWeb3.ts

```diff
@@ -119,3 +119,10 @@ export const getSDKWeb3Adapter = (signerAddress: string): Web3Adapter => {
     signerAddress,
   })
 }
+
+export const getSDKWeb3ReadOnly = (signerAddress: string): Web3Adapter => {
+  return new Web3Adapter({
+    web3: getWeb3ReadOnly(),
+    signerAddress,
```

Comment:
is signerAddress a required property? because the read only instance obv can't sign anything

Reply:
Let me test if I can remove it.

Reply:
It's a required param, but it happily works with a zero address, so I'll use that.